src/index.ts

```diff
@@ -107,6 +111,22 @@ const appConfig: AppConfig = {
 
 // Initialize and start the application
 const app = new App(appConfig);
+
+// Handle uncaught exceptions
+process.on("uncaughtException", (error: Error) => {
+	console.error("❌ Uncaught Exception:", error);
+	process.exit(1);
+});
+
+// Handle unhandled promise rejections
+process.on(
+	"unhandledRejection",
+	(reason: unknown, promise: Promise<unknown>) => {
+		console.error("❌ Unhandled Rejection at:", promise, "reason:", reason);
+		process.exit(1);
+	}
+);
```

Comment:
Worried about what logs could come out of this. 

Logging errors here might show private info in the logs. It's worth double-checking what gets printed so we don’t leak any secrets